Open Judge System/OJS.Services/Data/ExamGroups/ExamGroupsDataService.cs

```diff
@@ -37,5 +37,32 @@ public void Update(ExamGroup examGroup)
                 .Where(eg => eg.ExternalExamGroupId == externalId && eg.ExternalAppId == appId)
                 .Select(eg => eg.Id)
                 .FirstOrDefault();
+
+        public IQueryable<ExamGroup> GetAll() => this.examGroups.All();
+
+        public IQueryable<UserProfile> GetUsersByIdQuery(int id) =>
+            this.examGroups.All().Where(eg => eg.Id == id).SelectMany(eg => eg.Users);
+
+        public IQueryable<ExamGroup> GetByIdQuery(int id) =>
+            this.examGroups.All().Where(eg => eg.Id == id);
+
+        public void RemoveUserByIdAndUser(int id, string userId)
+        {
+            var examGroup = this.GetById(id);
+            var user = examGroup.Users.FirstOrDefault(u => u.Id == userId);
+            if (user != null)
+            {
+                examGroup.Users.Remove(user);
+                this.examGroups.SaveChanges();
+            }
+        }
+
+        public void RemoveReferencesFromContestByContest(int contestId) =>
```

Comment:
Why do we need the word `References` here? Why not simply `RemoveContestByContest` ?

Reply:
According to the convention of not repeating the name of the main model which the service is serving in its methods, `RemoveContestById` would seem as if we are removing a contest by the `ExamGroup` Id, whilst we are actually removing by Contest id.
According to the convention of adding the parameter by which we are performing the given operation we can't simply have `RemoveContest` as we are omitting the parameter from the method name.